fluster/image/raw_video.py

```diff
@@ -0,0 +1,104 @@
+# Fluster - testing framework for decoders conformance
+# Copyright (C) 2021, Fluendo, S.A.
+#  Author: Manuel Jimeno Martínez <mjimeno@fluendo.com>, Fluendo, S.A.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Library General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+from pathlib import Path
+from typing import Optional, List
+
+from fluster.image.raw_buffer import RawBuffer
+from fluster.image.raw_image import RawImage, get_meta_from_name
+
+
+class RawVideo:
+    """Class to handle the raw video"""
+
+    def __init__(
+        self,
+        buffer: bytes,
+        video_name: str,
+        width: int,
+        height: int,
+        buffer_type: RawBuffer.Type,
+    ):
+        self._buffer: bytes = buffer
+        self._video_name: str = video_name
+        self._width: int = width
+        self._height: int = height
+        self._buffer_type: RawBuffer.Type = buffer_type
+        self._frames = []
+
+        if buffer_type == RawBuffer.Type.YUV420:
+            if (width % 4) or (height % 4):
+                raise Exception("Width and height must be multiples of 4")
+            len_buffer = (width * height * 3) // 2
+        elif buffer_type == RawBuffer.Type.RGBA:
+            len_buffer = width * height * 4
+        else:
+            raise RuntimeError(f"The type buffer {buffer_type} is not supported")
+
+        # Slice the buffer using len_buffer
+        last_frame_start: int = 0
+        for next_frame_start in range(len_buffer, len(buffer) + 1, len_buffer):
+            self._frames.append(
```

Comment:
This is over-killing RAM. We can have a method to get a raw_frame at frame _i_, and extract that frame on demand.

Reply:
This is an eternal dilemma between performance and occupancy, it is designed to run a test, and it always requests all frames. So why should I generate them 1 to 1 if I can have them cached?

Reply:
Or you can create a cache of a with a maximum size in bytes, but that will be more difficult. However, this can be improved later, if you prefer that.